### ISSUE TITLE
JSONP callback data: array handling fails

### ISSUE THREAD

**Issue by slattery** [openseadragon/openseadragon#798]
Hello, and thanks for openseadragon.

I ran into an issue where I couldn't get an array of legacy tile configuration data to load from JSONP.   I am happy to start a proper fix and PR, but first I wanted to make sure that I wasn't committing gross user error in the first place, then talk about the scope of the problem and make sure I do something helpful.

__tl;dr__
The JSONP callback looks like it wants to treat the data as one tileSource, so I added a reprocessing of the incoming data, sending it to the Viewer's open() routine, in order to replace the tileSources config string with the JSON data.

__Symptoms__
If I passed in an array, the array wouldn't make it out of the callback, it would trip up on the determineType test with the "Unable to load TileSource" error on the Array.   (If I passed an unwrapped object, it would use the first entry then start using single letters of the js filename, which was amusing!)

For troubleshooting purposes, prior to the TileSource.determineType test, I added an array test under the string/parseXML test in the callback processor, then short-circuited back to 'window.viewer.open(data)' to reprocess the tileSource such that it would repopulate tileSources and then act on the first element, etc.

__Relevant troubleshooting scraps__
_inline config_
```js
var viewer = OpenSeadragon({
        id:                   "wat",
        sequenceMode:         true,
        crossOriginPolicy:    "Anonymous",
        autoHideControls:     0,
        immediateRender:      0,
        panHorizontal:        1,
        panVertical:          1,
        showNavigator:        1,
        showRotationControl:  1,
        prefixUrl:            '/sea/openseadragon/images/',
        tileSources:          "/sea/Records.js"
    });
```

_Records.js_ 
(Yes, there is only one level right now!  I  also shortened the URLs for cut-and-paste here.)

```js
Records(
[{ type: 'legacy-image-pyramid', levels: [{ url:'http://s3.amazonaws.com/Record_00a.jpg', width:750, height:973}]},
{ type: 'legacy-image-pyramid', levels: [{ url:'http://s3.amazonaws.com/Record_00b.jpg', width:750, height:973}]},
{ type: 'legacy-image-pyramid', levels: [{ url:'http://s3.amazonaws.com/Record_001.jpg', width:750, height:973}]}
...
])
```

_Simple hack_  
I added this lines in my copy of the pre-built version:
```text
//! OpenSeadragon 2.1.0
//! Built on 2015-11-12
//! Git commit: v2.1.0-3-b2c17b5
```
This comment corresponds to [/src/tilesource.js#L405-409](https://github.com/openseadragon/openseadragon/blob/d1afae6ca0098b82f7c7dd8e9019806ffe228421/src/tilesource.js#L405-L409) in the source.

I added an array check for data under the string check for data (the one that guesses for XML), if found and the original tileSources value is a string, then we're guessing we arrived here via JSONP and send the array back up to window.viewer.open(data) to make sure that tileSources is populated with the data array and to restart processing the tileSources elements.

```js
 else if ( $.isArray(data) && typeof(window.viewer.tileSources) === "string") {
              return window.viewer.open(data); 
 } 

```

**Comment by iangilman** [comment]
Good point. I suppose at the very least we should identify that it's an array and give a more meaningful error. 

I'm hesitant to just pass the data straight into `open`, since it's such a dynamic function anyway, I'd be concerned about security issues or other unintended consequences. I think I'd at least want to make sure all of the items in the array were just tile sources and not TiledImage specs, to cut down on the possibility space. 

Note that this issue can also be addressed outside of OSD by loading the JSON and then passing it in, like so (assuming jQuery): 

```
$.get('/sea/Records.js', function(data) {
    var viewer = OpenSeadragon({
        tileSources: data,
        ...
    });    
}, 'jsonp');
```

**Comment by slattery** [comment]
Agreed that it is quite hacky to stuff the array back into `open`.      I'll change the array check routine to generate an error instead of the re-route.    Thanks for your reply.

**Comment by iangilman** [comment]
Sounds good. Thank you for diving in!
